### PR TITLE
Add C2D_RELEASE env and Consul check version.

### PR DESCRIPTION
**Pull request by aav66**

### CODE REVIEW

File: templates/Dockerfile.template [hunk 1]
```diff
@@ -76,8 +78,9 @@ RUN mkdir -p /tmp/build && \
     unzip -d /bin consul_${CONSUL_VERSION}_linux_amd64.zip && \
     cd /tmp && \
     rm -rf /tmp/build && \
-# tiny smoke test to ensure the binary we downloaded runs
-    consul version
+# tiny smoke test to ensure the binary we downloaded runs right version
+    test ${C2D_RELEASE} = `consul version | awk '/Consul/ { print $2 }' | grep -o -E '([0-9]+\.)([0-9]+\.)([0-9]+)'$` \
```
Comment (KadharNiminen) [inline templates/Dockerfile.template]: $(var)
fix the execution

Comment (aav66) [reply templates/Dockerfile.template]: Fixed

File: 1/1.4/Dockerfile [hunk 2]
```diff
@@ -3,6 +3,8 @@ FROM gcr.io/google-appengine/debian9:latest
 # This is the release of Consul to pull in.
 ENV CONSUL_VERSION=1.4.4
 
+ENV C2D_RELEASE=${CONSUL_VERSION}
```
Comment (wgrzelak) [inline 1/1.4/Dockerfile:6]: Can we always have a static value assigned to the `C2D_RELEASE` variable, so we are sure what the result of the variable is.

For example: Currently if the `CONSUL_VERSION` will be modified to `1.4`, then we may not catch that `C2D_RELEASE` is incorrect during the review.

**Comment by KadharNiminen** [comment]
`gpg --keyserver $server --recv-keys $GOSU_GPG` -> $(gpg --keyserver $server --recv-keys $GOSU_GPG)

File: 1/1.4/Dockerfile [hunk 3]
```diff
@@ -87,8 +89,9 @@ RUN mkdir -p /tmp/build && \
     unzip -d /bin consul_${CONSUL_VERSION}_linux_amd64.zip && \
     cd /tmp && \
     rm -rf /tmp/build && \
-# tiny smoke test to ensure the binary we downloaded runs
-    consul version
+# tiny smoke test to ensure the binary we downloaded runs right version
+    test ${C2D_RELEASE} = $(consul version | awk '/Consul/ { print $2 }' | grep -o -E '([0-9]+\.)([0-9]+\.)([0-9]+)'$) \
+         || echo "Version check failed"
```
Comment (wgrzelak) [inline 1/1.4/Dockerfile]: If the test failure then the result is printing the message with `0` exit code.

```suggestion
         || { echo "Version check failed"; exit 1; }
```

File: 1/1.4/Dockerfile (hunk 2 above)
Comment (aav66) [reply 1/1.4/Dockerfile:6]: As i remember we have agreed to use C2D_RELEASE to get version of app in v2 Makefile. 
Also we can write simple test in Makefile v2 to be sure C2D_RELEASE starts with TRACK and not match the ^TRACK$ 
What do you think?

File: 1/1.4/Dockerfile (hunk 3 above)
Comment (aav66) [reply 1/1.4/Dockerfile]: Thanks for suggestions!

Comment (aav66) [reply 1/1.4/Dockerfile]: Fixed

File: 1/1.4/Dockerfile (hunk 2 above)
Comment (wgrzelak) [reply 1/1.4/Dockerfile:6]: We can run tests, but maybe this is overkill as we already have smoke test in the Dockerfile.

I still would like to have static value here.

In go template it should be:

```
ENV C2D_RELEASE={{ $consul.Version }}
```

**Review by wgrzelak — APPROVED** [review]

**Pull request merged**